docs/dsl_inspec.md

```diff
@@ -239,6 +239,43 @@ end
 
 Mixing this with other conditionals (like checking existence of the files etc.) can help to test different test paths using InSpec. This way you can skip certain tests which would 100% fail due to the way servers are prepared, but you know that the same test suites are reused later in different circumstances by different teams.
 
+Some notes about `only_if`:
+
+ * `only_if` applies to the entire `control`. If the results of the `only_if` block evaluate to false, the contents of the describe blocks will not be run. However, the describe block resources (`command('redis-cli SET test_inspec "HELLO"')` in the above example) that preceded the only_if WILL run. For this reason, `only_if` should usually come first in the control, prior to any `describe` blocks.
```

Comment:
I don't think this is true.
I wrote this test control - 

```
control 'only_if test' do

  describe command('echo \'AFTER\' > /tmp/AFTER') do
    its('exit_status') { should cmp 0 }
  end

  only_if { false }

end
```

The command causes the file `/tmp/AFTER` to exist. When I run this, the control is skipped and the file does not exist. If I change the `false` to `true` then the control runs and the file is created.

A case where this is true is where Ruby statements are placed outside of InSpec's describe/its DSL:

```
control 'only_if test' do

  result = command('echo \'AFTER\' > /tmp/AFTER').exit_status

  describe result do
    it { should cmp 0 }
  end

  only_if { false }

end
```

For this control, the tmp file WILL get created.

Keka, @Schwad and myself have been discussing how Ruby statements like this are generally bad practice because they execute regardless of skip/waiver/exclude_control etc etc. If warning the user here about Ruby statements it would be good to make clear that they are generally not good practice (not just for this specific case).

Reply:
I see the test case is doing `describe 1/0 do`
There, the `1/0` is a Ruby evaluation.

Reply:
Well, you're right-ish. It's at least sort of true :-) The fact that 1/0 is evaluated in a preceding control, and not in a subsequent control, indicates that there is a difference.

I have no idea how that works.

But I tried it with a command(echo) and I got the same result as you. I'll edit accordingly and treat the ruby expression case as an undocumented edge case.